### PR TITLE
add option no dereference.

### PR DESCRIPTION
**Pull request by vwxyzh**

### CODE REVIEW

File: src/docfx/Models/BuildCommandOptions.cs [hunk 1]
```diff
@@ -123,7 +123,10 @@ internal class BuildCommandOptions : ICanPrintHelpMessage, ILoggable
         [OptionList("postProcessors", Separator = ',', HelpText = "Set the order of post processors in plugins")]
         public List<string> PostProcessors { get; set; }
 
-        [OptionList("lruSize", HelpText = "Set the LRU cached model count (approximately the same as the count of input files). By default, it is 8192 for 64bit and 3072 for 32bit process. With LRU cache enabled, memory usage decreases and time consumed increases. If set to 0, Lru cache is disabled.")]
+        [Option("lruSize", HelpText = "Set the LRU cached model count (approximately the same as the count of input files). By default, it is 8192 for 64bit and 3072 for 32bit process. With LRU cache enabled, memory usage decreases and time consumed increases. If set to 0, Lru cache is disabled.")]
         public int? LruSize { get; set; }
+
+        [Option("noDereference", HelpText = "If set to true, manifest will keep links form.")]
```
Comment (hellosnow) [inline src/docfx/Models/BuildCommandOptions.cs]: >links form [](start = 80, length = 10)

minor: add explanation for what's "links from"? how about "keep original links"?

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs [hunk 2]
```diff
@@ -169,7 +168,22 @@ public void Build(IList<DocumentBuildParameters> parameters, string outputDirect
...
+                        if (parameters[0].NoDereference)
+                        {
+                            var count = (from f in generatedManifest.Files
+                                         from o in f.OutputFiles
+                                         select o.Value into v
+                                         where v.LinkToPath != null
+                                         select v).Count();
+                            if (count > 0)
+                            {
+                                Logger.LogInfo($"Skip dereference for {count} files.");
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs]: Skip *dereferencing*

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuildParameters.cs [hunk 3]
```diff
@@ -66,6 +66,9 @@ public sealed class DocumentBuildParameters : MarshalByRefObject
         [IncrementalIgnore]
         public int LruSize { get; set; }
 
+        [IncrementalIgnore]
+        public bool NoDereference { get; set; }
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuildParameters.cs]: naming: DisableDereference?

File: src/docfx/Models/BuildCommandOptions.cs (hunk 1 above)
Comment (vicancy) [inline src/docfx/Models/BuildCommandOptions.cs]: How about: If set to true, for output files that are kept unchanged from last build, docfx does not deference (aka. copy) them to the output folder, instead, it saves a `LinkFrom` property inside `mainfiest.json` to indicate the physical location of that file.

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuildParameters.cs (hunk 3 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuildParameters.cs]: `KeepFileLink`?

**Pull request merged**